pkg/cvo/upgradeable.go

```diff
@@ -234,6 +247,126 @@ func (check *clusterManifestDeleteInProgressUpgradeable) Check() *configv1.Clust
 	return nil
 }
 
+func gateApplicableToCurrentVersion(gateName string, currentVersion string) (bool, error) {
+	var applicable bool
+	if ackVersion := adminAckGateRegexp.FindString(gateName); ackVersion == "" {
+		return false, fmt.Errorf("%s configmap gate name %s has invalid format; must begin with 'ack-X.Y'",
+			internal.AdminGatesConfigMap, gateName)
+	} else {
+		parts := strings.Split(ackVersion, "-")
+		if strings.HasPrefix(currentVersion, parts[1]) {
+			applicable = true
+		}
+	}
+	return applicable, nil
+}
+
+func checkAdminGate(gate string, currentVersion string,
+	gateConfigmap *corev1.ConfigMap, ackConfigmap *corev1.ConfigMap,
+	ackConfigmapErr error) *configv1.ClusterOperatorStatusCondition {
+
+	if applies, err := gateApplicableToCurrentVersion(gate, currentVersion); err == nil {
+		if !applies {
+			return nil
+		}
+	} else {
+		klog.Error(err)
+		return &configv1.ClusterOperatorStatusCondition{
+			Type:    configv1.ClusterStatusConditionType("UpgradeableAdminAckRequired"),
+			Status:  configv1.ConditionFalse,
+			Reason:  "AdminAckConfigMapIssue",
+			Message: err.Error(),
+		}
+	}
+	var errorMessage string
+	var gateMessage string
+	if val, ok := gateConfigmap.Data[gate]; !ok {
+		errorMessage = fmt.Sprintf("%s configmap does not contain gate %s.", internal.AdminGatesConfigMap, gate)
+	} else if val == "" {
+		errorMessage = fmt.Sprintf("%s configmap gate %s has no value.", internal.AdminGatesConfigMap, gate)
```

Comment:
nit: "no value" -> "an empty value"?  Or "is not $WHATEVER_WE_NEED_FOR_AN_ACK"?

Reply:
Oh, wait, this is the thing we control.  Probably want some different reason for that, because it should never happen, and we want it to be really obvious in Telemetry if that's going on.

Reply:
Changed to "...must contain a non-empty value".

Reply:
Didn't see your second comment until after the change above. Right, it should not in the field since this will cause upgrade CI test to fail once this blocker is active (in 4.8).